### PR TITLE
Fix suggestion to add unneeded space in `manual_async`

### PR DESCRIPTION
**Pull request by giraffate**
Fix a same case as https://github.com/rust-lang/rust-clippy/pull/6247

changelog: Fix suggestion to add unneeded space in `manual_async`

### CODE REVIEW

File: clippy_lints/src/manual_async_fn.rs [hunk 1]
```diff
@@ -69,7 +69,20 @@ impl<'tcx> LateLintPass<'tcx> for ManualAsyncFn {
                     |diag| {
                         if_chain! {
                             if let Some(header_snip) = snippet_opt(cx, header_span);
-                            if let Some(ret_pos) = header_snip.rfind("->");
+                            if let Some(ret_pos) = header_snip.rfind("->").map(|rpos| {
+                                let mut rpos = rpos;
+                                let chars: Vec<char> = header_snip.chars().collect();
```
Comment (Manishearth) [inline clippy_lints/src/manual_async_fn.rs:74]: could you refactor this into something for `utils` so https://github.com/rust-lang/rust-clippy/pull/6247/ can also use it

**Review by Manishearth — APPROVED** [review]
code seems fine, would like it to be a common function

File: clippy_lints/src/manual_async_fn.rs (hunk 1 above)
Comment (giraffate) [reply clippy_lints/src/manual_async_fn.rs]: Thanks for your review! I refactored this.

**Comment by Manishearth** [comment]
@bors r+

i was wondering what a good name for the function would be but this is as good as any

**Pull request merged**